tests/project_helpers.py

```diff
@@ -165,22 +186,23 @@ def __call__(self, cache: pytest.Cache) -> Path:
         script or the requirements to create that venv change.
         """
         # We cache venv dirs using the hash from create_venv_hash
-        cached_str = cache.get(f"fawltydeps/{self.venv_hash()}", None)
+        cached_str = cache.get(os.path.join("fawltydeps", f"{self.venv_hash()}"), None)
         if cached_str is not None and Path(cached_str, ".installed").is_file():
             return Path(cached_str)  # already cached
 
         # Must run the script to set up the venv
         venv_dir = Path(cache.mkdir(f"fawltydeps_venv_{self.venv_hash()}"))
         logger.info(f"Creating venv at {venv_dir}...")
         venv_script = self.venv_script_lines(venv_dir)
+
```

Comment:
With the rewritten `venv_script_lines()`, this becomes:
```suggestion
        venv_script = list(self.venv_script_lines(venv_dir))

```